server/task_check_scheduler.go

```diff
@@ -177,8 +177,90 @@ func (s *TaskCheckScheduler) Register(taskType string, executor TaskCheckExecuto
 	s.executors[taskType] = executor
 }
 
+// shouldScheduleTimingTaskCheck will return whether should we schedule a timing task check for current task
+// the logic goes like the following:
+// 1. user specified the notBeforeTs field, hence a timing gate is added
+// 2. there is no running timing task check
+// 3(a). the notBeforeTs has altered since last check (either creation or patch)
+// 3(b). the notBeforeTs has NOT altered since last check && last check had failed && notBeforeTs has passed now
```

Comment:
the noteBeforeTs has NOT been altered since last check, however the last check had failed && notBeforeTs has passed now.